_posts/2016-10-24-deploy-jekyll-with-travis.md

```diff
@@ -0,0 +1,232 @@
+
+### Set up a GitHub token for Travis
+
+To give Travis the authority to commit and push to our repository, we generated
+a personal access token by visiting the
+[settings page](https://github.com/settings/tokens) on GitHub and generating a
+new token. We used a bot account that has access to our repo to avoid needing to
+use one of our personal accounts.
+
+![Generating a personal access token in GitHub]({{ site.url }}/assets/img/blog/github-token.jpg)
+
+We'll need to use the access token and the account's email address in our
+deployment script, so to keep those items out of version control we used
+`travis encrypt`:
+
+```bash
+$ travis encrypt GH_TOKEN=<token> --add env.global
+$ travis encrypt GH_EMAIL=<email> --add env.global
+```
+
+This adds encrypted versions of these sensitive items to `travis.yml`:
+
+```yaml
+env:
+  global:
+  - secure: [big long encrypted thing]
+  - secure: [and another]
+```
+
+With that, we can use the `GH_TOKEN` and `GH_EMAIL` variables in our
+deployment script.
+
+### Configure deployment
+
+Next we added the following to our `travis.yml` file:
+
+```yaml
+# Keep Travis from testing `master` branch
+branches:
+  except:
+  - master
+
+# Deployment config
+deploy:
+  provider: script
+  script: "./_scripts/build.sh"
+  skip_cleanup: false
+  on:
+    branch: source
+```
+
+We don't want Travis running our tests on the `master` branch anymore since it
+only contains the compiled site, and only want Travis to run the
+deployment script on the `source` branch. We're also telling Travis which script
+to run on deployment, and to destroy the site after the test script is run with our
+`config.test.yml` file so the deployment script can rebuild the site with our
+default `config.yml` file. Check out the [full travis.yml file here](https://github.com/savaslabs/savaslabs.github.io/blob/95bb1d70790acf4c33b7c121e2ff462af5096d73/.travis.yml).
+
+That version of `travis.yml` worked for us when we were using `jekyll build` to
+compile the site, but things got trickier when we started using gulp. We had to
+set the proper versions of npm and node.js, and ensure that our ruby gems were
+installed properly. Here's all the travis config we needed to get things
+running with gulp:
+
+```yaml
+# travis.yml
+
+language: node_js
+sudo: required
+script:
+  - ./_scripts/run-tests.sh
+branches:
+  except:
+  - master
+node_js:
+ - '6.1'
+before_install:
+  - if [[ `npm -v` != 3* ]]; then npm i -g npm@3; fi
+  - rvm install 2.1
+  - rvm use 2.1
+  - . $HOME/.nvm/nvm.sh && nvm install 6.1 && nvm use 6.1
+  - gem install bundler
+  - bundle check || bundle install
+env:
+  global:
+  - NOKOGIRI_USE_SYSTEM_LIBRARIES=true
+  - secure: [stuff]
+  - secure: [more stuff]
+deploy:
+  provider: script
+  script: "./_scripts/build.sh"
+  skip_cleanup: false
+  on:
+    branch: source
+```
+
+### The deployment script
+
+When the `source` branch is updated and our tests pass, the deployment script is
+executed.
```

Comment:
I think it'd be helpful to add the script's file name for clarity like you do in the other sections
> ...the deployment script `build.sh` is executed.